File: src/main/java/ru/tykvin/hermes/configuration/StorageConfiguration.java
package ru.tykvin.hermes.configuration;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;

import java.time.Duration;

@Getter
@Setter
@RequiredArgsConstructor
@Configuration
@ConfigurationProperties("storage")
public class StorageConfiguration {
    private String root;
    private String host;
    private Duration lifeTime;
}

File: src/main/kotlin/ru/tykvin/hermes/configuration/StorageConfiguration.kt
package ru.tykvin.hermes.configuration

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component

import java.time.Duration

@Component
@ConfigurationProperties("storage")
class StorageConfiguration {
    lateinit var root: String
    lateinit var host: String
    lateinit var lifeTime: Duration
}
